internal/eventlog/eventlog.go

```diff
@@ -18,12 +18,13 @@ type Event struct {
 func NewEvent(t time.Time, name string) *Event {
 	return &Event{
 		Name:      name,
+		Outcome:   enums.EventOutcomeUnspecified,
 		StartedAt: t,
 	}
 }
 
-func (e *Event) IsValid() bool {
-	return e.Outcome == enums.EventOutcomeValid
+func (e *Event) IsSuccess() bool {
+	return e.Outcome == enums.EventOutcomeSuccess
 }
 
 func (e *Event) Complete(t time.Time, outcome enums.EventOutcome, msg string, a ...any) *Event {
```

Comment:
`a` param is no longer needed after the changes in the wrapper, and `msg` should be already formatted and can be used directly to set the message value.

Reply:
Huh, interesting. I don't have the same aversion that you do to using `any` in a typed language, but I can understand why you dislike it.  I do think `fmt.Sprintf()` is a good example of a function that demonstrates why allowing an `any` parameter type is necessary in a typed GPL.

I did notice while writing the code that the hints from `fmt.Sprintf()` bubble up to the wrapper functions, which _is_ a bit magical, but I think it's magical in a good way. :)